src/functions/ducklake_get_option.cpp

```diff
@@ -0,0 +1,284 @@
+#include "functions/ducklake_table_functions.hpp"
+#include "storage/ducklake_transaction.hpp"
+#include "storage/ducklake_catalog.hpp"
+#include "storage/ducklake_schema_entry.hpp"
+#include "storage/ducklake_table_entry.hpp"
+#include <unordered_map>
+
+namespace duckdb {
+
+struct OptionMetadata {
+	string description;
+	string type;
+};
+
+static constexpr const char *OPTION_TYPE_USER = "user";
+static constexpr const char *OPTION_TYPE_SYSTEM = "system";
+static constexpr const char *DEFAULT_OPTION_DESCRIPTION = "Custom configuration option";
+
+static constexpr const char *COLUMN_NAME = "name";
+static constexpr const char *COLUMN_VALUE = "value";
+static constexpr const char *COLUMN_TYPE = "type";
+static constexpr const char *COLUMN_DESCRIPTION = "description";
+
+static const std::unordered_map<string, OptionMetadata> DUCKLAKE_OPTIONS = {
```

Comment:
In general we avoid static C++ objects because of destructor order - can we turn this into a C-style array instead (see e.g. `src/common/ducklake_types.cpp`)?